### PR TITLE
Add CSS spinner

### PR DESCRIPTION
**Pull request by benfrankel**
Fixes https://github.com/TheBevyFlock/bevy-template/issues/8.

Using the bevy logo colors:

2024-07-08_1720433342_320x284

### CODE REVIEW

**Comment by janhohenheim** [comment]
I'd be happy if my local coding wizard @bash could review this for me. _summoning noises_

**Comment by MiniaczQ** [comment]
Poor Tau gets work thrown their way,
after this gets merged lets run a release and have this on itch.io together with splash screen

File: web/index.html [hunk 1]
```diff
@@ -2,25 +2,42 @@
...
+        // Hide loading screen when the game starts.
+        const loading_screen = document.getElementById('loading-screen');
+        const bevy = document.getElementById('bevy');
+        const observer = new MutationObserver(() => {
+            if (bevy.height > 1) {
+                loading_screen.style.display = 'none';
+                observer.disconnect();
+            }
+        });
+        observer.observe(bevy, { attributes: true });
```
Comment (bash) [inline web/index.html:39]: Nitpick: you only care about the height change, so no need to watch for all changes:
```suggestion
        observer.observe(bevy, { attributeFilter: ['height'] });
```

File: web/style.css [hunk 2]
```diff
@@ -8,4 +8,44 @@ html,
...
+    width: 100%;
+    height: 100%;
+    display: flex;
+    justify-content: center;
+    align-items: center;
+    flex-direction: column;
+}
+
+#loading-screen {
+    background-color: #282828;
```
Comment (bash) [inline web/style.css:23]: This color should match the in-game splash screen color, right?

I'd put that in a nice css variable at the top so it's extra clear that this needs to be touched by consumers of the template.

```css
:root {
    --splash-screen-bg-color: #282828;
}

/* ... */

#loading-screen {
    background-color: var(--splash-screen-bg-color);
}
```

**Review by bash — CHANGES_REQUESTED** [review]
Looks good to me otherwise ✨

File: web/index.html [hunk 3]
```diff
@@ -2,25 +2,42 @@
...
+            if (!error.message.startsWith("Using exceptions for control flow, don't mind me. This isn't actually an error!")) {
+                throw error;
+            }
+        });
+
+        // Hide loading screen when the game starts.
+        const loading_screen = document.getElementById('loading-screen');
+        const bevy = document.getElementById('bevy');
+        const observer = new MutationObserver(() => {
+            if (bevy.height > 1) {
```
[the remark this answers is not in the document]
Comment (benfrankel) [reply web/index.html:34]: Yep!

File: web/index.html (hunk 1 above)
Comment (benfrankel) [reply web/index.html]: Neat.

File: web/style.css (hunk 2 above)
Comment (benfrankel) [reply web/style.css:23]: I hadn't considered it, but it's probably good if it matches, yeah.

**Review by bash — APPROVED** [review]

**Review by janhohenheim — APPROVED** [review]

**Comment by MiniaczQ** [comment]
I'm trusting Jan and Bash on this one

**Comment by janhohenheim** [comment]
I'm trusting @bash on this one

**Comment by benfrankel** [comment]
It's okay. You can all trust me on this one.

**Comment by janhohenheim** [comment]
Sounds legit to me!

**Pull request merged**